karapace/config.py

```diff
@@ -88,17 +90,45 @@ def parse_env_value(value: str) -> Union[str, int, bool]:
 
 
 def set_config_defaults(config: Config) -> Config:
-    for k, v in DEFAULTS.items():
-        if k.startswith("karapace"):
-            env_name = k.upper()
-        else:
-            env_name = f"karapace_{k}".upper()
-        if env_name in os.environ:
-            val = os.environ[env_name]
-            print(f"Populating config value {k} from env var {env_name} with {val} instead of config file")
-            config[k] = parse_env_value(os.environ[env_name])
-        config.setdefault(k, v)
+    new_config = DEFAULTS.copy()
+    new_config.update(config)
 
+    set_settings_from_envionment(new_config)
+    set_sentry_dsn_from_environment(new_config)
+    validate_config(new_config)
+    return new_config
+
+
+def set_settings_from_envionment(config: Config):
```

Comment:
```suggestion
def set_settings_from_environment(config: Config):
```

typo in function name